File: src/main/java/com/college/hotlittlepigs/user/dto/UserSaveDTO.java
package com.college.hotlittlepigs.user.dto;

import com.college.hotlittlepigs.user.User;
import lombok.Data;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

@Data
public class UserSaveDTO {

  @NotBlank(message = "Name is required")
  private String name;

  @Email(message = "Invalid email address")
  private String email;

  @Size(min = 8, max = 16, message = "Password size must be between 8 and 16")
  private String password;

  public User toUser() {
    return new User(null, this.name, this.email, this.password, null, null);
  }
}

File: src/main/kotlin/com/college/hotlittlepigs/user/dto/UserSaveDTO.kt
package com.college.hotlittlepigs.user.dto

import com.college.hotlittlepigs.user.User
import javax.validation.constraints.Email
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.Size

class UserSaveDTO {
    @NotEmpty
    lateinit var name: String

    @Email
    lateinit var email: String

    @Size(min = 8, max = 16)
    lateinit var password: String

    fun toUser(): User = User(name, email, password)
}